### PR TITLE
Better OpenGL detection

### PR DESCRIPTION
**Pull request by mrkubax10**
This PR makes VideoSystem use OpenGL version information reported by OS when determining which OpenGL version to use.

### CODE REVIEW

**Comment by lorn10** [comment]
I mention that here too, - this "better OpenGL fallback" really works in my case. :+1: However, this is true for the **ubuntu-latest-64-gcc-Release-appimage** but not the **glbinding** version. More information can be found at my original bug report #2173.

**Comment by Semphriss** [comment]
[Relevant:](https://stackoverflow.com/questions/25171460/which-version-does-glgetstringgl-version-get)

```cpp
GLint opengl_major_version;
glGetIntegerv(GL_MAJOR_VERSION, &opengl_major_version);
if (opengl_major_version > 3) {
 ...
```

**Comment by mrkubax10** [comment]
According to this website: https://www.khronos.org/opengl/wiki/OpenGL_Context
> `glGetIntegerv(GL_MAJOR_VERSION, *);`
> `glGetIntegerv(GL_MINOR_VERSION, *);`
> These are available on OpenGL version 3.0 and above contexts. ...

**Comment by lorn10** [comment]
Again a short side-note. Regarding the **Mesa** drivers on Linux this is absolutely no problem, the **GL_ARB_compatibility** feature is available. :+1: 

However, this may be an issue for (older?) Mac OS versions prior 10.7, see [OpenGL Capabilities Tables - Apple Developer](https://developer.apple.com/opengl/OpenGL-Capabilities-Tables.pdf).

File: src/video/gl/gl_video_system.cpp [hunk 1]
```diff
@@ -61,7 +61,27 @@ GLVideoSystem::GLVideoSystem(bool use_opengl33core) :
...
+    sscanf(version_string, "%d", &major);
+
+    if (major >= 3)
+      m_use_opengl33core = true;
+    else if (major == 2)
+      m_use_opengl33core = false;
+    else
+    {
+      // OpenGL 2.0 or higher is unsupported, throw exception so SDL renderer will be used instead
+      throw std::exception();
```
Comment (Semphriss) [inline src/video/gl/gl_video_system.cpp:78]: There should be a message here that explains what happened

**Review by weluvgoatz — APPROVED** [review]

**Review by Zwatotem — APPROVED** [review]

**Pull request merged**